### PR TITLE
HPCC-18356 Build break on latest Clang

### PR DESCRIPTION
**Pull request by richardkchapman**
Recoded using modern C++ constructs in place of variadic function.

Note there are warnings in hql for similar issues, but they have not been
turned into errors (yet) so we don't HAVE to fix them (yet).

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
Compiled...

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

File: system/jlib/jstats.cpp [hunk 1]
```diff
@@ -1086,39 +1086,21 @@ static int compareUnsigned(unsigned const * left, unsigned const * right)
...
-    {
-        unsigned next  = va_arg(args, unsigned);
-        if (!next)
-            break;
-        indexToKind.appendUniq(next);
-    }
-    va_end(args);
+    for (auto kind : kinds)
+        if (kind != StKindNone)
+            indexToKind.append(kind);
```
Comment (ghalliday) [inline system/jlib/jstats.cpp]: Should probably be appendUnique - or createMapping should throw an exception if there is a duplicate mapping.

**Comment by ghalliday** [comment]
A neat change.  We should probably change our arrays so they are also compatible with the range base for syntax.

File: system/jlib/jstats.cpp [hunk 2]
```diff
@@ -1086,39 +1086,21 @@ static int compareUnsigned(unsigned const * left, unsigned const * right)
...
-            unsigned next  = va_arg(args, unsigned);
-            if (!next)
-                break;
-            indexToKind.appendUniq(next);
-        }
-        va_end(args);
-    }
+    for (auto kind : kinds)
+        if (kind != StKindNone)
+            indexToKind.append(kind);
```
Comment (ghalliday) [inline system/jlib/jstats.cpp]: Should also be appendUnique.  Possible would be better too assert that it isn't StKindNone.

**Pull request merged**